testbed/core/views.py

```diff
@@ -72,3 +82,200 @@ def index(request):
         'destination_actor': user_actors.filter(role=Actor.ROLE_DESTINATION).first(),
         'oauth_form': oauth_form,
     })
+
+
+# OAuth Testing Views
+
+def oauth_callback(request):
+    """
+    Handle the callback from the OAuth server
+    This displays the authorization code or error message
+    """
+    # Get query parameters
+    code = request.GET.get('code')
+    error = request.GET.get('error')
+    error_description = request.GET.get('error_description')
+    state = request.GET.get('state')
+    
+    # Initialize context with received parameters
+    context = {
+        'code': code,
+        'error': error,
+        'error_description': error_description,
+        'state': state
+    }
+    
+    # Validate the state parameter to prevent CSRF attacks
+    if not state:
+        logger.warning("OAuth callback received with no state parameter")
+        context['error'] = 'invalid_state'
+        context['error_description'] = 'No state parameter provided'
+    else:
+        is_valid_state = validate_state_from_session(request, state)
```

Comment:
Does this save the authorization token to the DB? if so would be good to comment here so we know where/when the authorization token comes in.